CMake build structure and style and pkg-config fixes

eaaltonen opened the pull request:
Re-opened after #350 was closed.

Resolve some inconsistencies originating from commit https://github.com/OP-TEE/optee_client/commit/e0a12e2322ae02e957910f29948bd19455bf9cfa "cmake: Initial CMake support".

I think these would be good to fix in order to:

* improve the consistency
* prepare for adding a CMake config-file package to the install
* install the `teec` pkgconfig file

jforissier reviewed (commented):
For "cmake: pkgconfig file: install the file":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

eaaltonen commented:
Squeashed fixup.

>  For "cmake: pkgconfig file: install the file":
> 
> Acked-by: Jerome Forissier <jerome.forissier@linaro.org>

added.

> I believe include/linux/tee.h and include/teec_benchmark.h should be moved under src. They are not part of the libteec API.

Moved, thanks.

jforissier reviewed (commented):
Thanks. For "Makefile, cmake: move teec related headers":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

etienne-lms reviewed (commented):
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the whole series.

eaaltonen commented:
`Acked-by`, thanks.

eaaltonen commented on CMakeLists.txt:
```diff
@@ -40,7 +40,7 @@ endif(CCACHE_FOUND)
 add_subdirectory(libteec)
 add_subdirectory(tee-supplicant)
-add_subdirectory(public)
+
```
I'll remove this still.
